util/util.go

```diff
@@ -236,3 +238,32 @@ func CheckBackupType(backupTarget string) (string, error) {
 
 	return u.Scheme, nil
 }
+
+func ResolveBackingFilepath(fileOrDirpath string) (string, error) {
+	fileOrDir, err := os.Open(fileOrDirpath)
+	if err != nil {
+		return "", err
+	}
+	defer fileOrDir.Close()
+
+	fileOrDirInfo, err := fileOrDir.Stat()
+	if err != nil {
+		return "", err
+	}
+
+	if fileOrDirInfo.IsDir() {
+		files, err := fileOrDir.Readdir(-1)
+		if err != nil {
+			return "", err
+		}
+		if len(files) != 1 {
+			return "", fmt.Errorf("expected exactly one file, found %d files/subdirectories", len(files))
+		}
+		if files[0].IsDir() {
+			return "", errors.New("expected exactly one file, found a subdirectory")
```

Comment:
I think you missed the comment above. You don't need to import `errors` just for `errors.New` since you can use `fmt.Errorf`.

Reply:
yes, my bad